### PR TITLE
Create an issue on GitHub Actions error

### PR DESCRIPTION
**Pull request by danielrozenberg**
@-mentioning...

release-on-duty:
* promote-*.yml
* sync-versions-to-cdn.yml

build-on-duty:
* sync-client-side-experiments-to-cdn.yml

wg-infra:
* auto-approve.yml

No issue for:
* cherry-pick.yml
* ci.yml

Example of an issue that was created from a failed workflow: https://github.com/danielrozenberg/cdn-configuration/issues/34

### CODE REVIEW

File: .github/WORKFLOW_ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: 'Workflow "{{ env.WORKFLOW_NAME }}" failed (run_id: {{ env.RUN_ID }})'
+---
+
+The workflow "**{{ env.WORKFLOW_NAME }}**" failed. See logs:
+https://github.com/{{ env.REPO_SLUG }}/actions/runs/{{ env.RUN_ID }}
+
+// cc: {{ env.MENTION }}
```
Comment (estherkim) [inline .github/WORKFLOW_ISSUE_TEMPLATE.md:8]: Should we assign the issue instead/as well as mention?

File: .github/WORKFLOW_ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (estherkim) [inline .github/WORKFLOW_ISSUE_TEMPLATE.md:1]: Hmm, this doesn't really follow GitHub issue template syntax... i.e. it's supposed to be a `.yml` file?

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository

File: .github/WORKFLOW_ISSUE_TEMPLATE.md (hunk 1 above)
Comment (estherkim) [reply .github/WORKFLOW_ISSUE_TEMPLATE.md:8]: And add a reminder to close the issue when resolved.

**Review by estherkim — COMMENTED** [review]
Can you create an issue for the cherry-pick workflow too?

File: .github/workflows/sync-client-side-experiments-to-cdn.yml [hunk 3]
```diff
@@ -30,3 +30,10 @@ jobs:
         env:
           CF_ACCOUNT_ID: ${{ secrets.CF_ACCOUNT_ID }}
           CF_API_TOKEN: ${{ secrets.CF_API_TOKEN }}
+
+  create-issue-on-error:
+    needs: sync
+    if: failure()
+    uses: ampproject/cdn-configuration/.github/workflows/create-issue-on-error.yml@main
+    with:
+      mention: '@ampproject/build-on-duty'
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply .github/workflows/sync-client-side-experiments-to-cdn.yml]: Done 2311cc4

File: .github/workflows/promote-beta-experimental-opt-in.yml [hunk 4]
```diff
@@ -29,3 +29,10 @@ jobs:
       auto-merge: ${{ github.event.inputs.auto-merge }}
     secrets:
       access-token: ${{ secrets.ACCESS_TOKEN }}
+
+  create-issue-on-error:
```
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply .github/workflows/promote-beta-experimental-opt-in.yml:33]: lol I knew there was something I wanted to try and forgot to... welp, turns out the answer is no:
https://github.com/danielrozenberg/cdn-configuration/actions/runs/1763098096/workflow

> ```error parsing called workflow "danielrozenberg/cdn-configuration/.github/workflows/promote-reusable-workflow.yml@main": job "create-issue-on-error" calls workflow "danielrozenberg/cdn-configuration/.github/workflows/create-issue-on-error.yml@main", but doing so would exceed the limit on called workflow depth of 2```

File: .github/WORKFLOW_ISSUE_TEMPLATE.md (hunk 1 above)
Comment (danielrozenberg) [reply .github/WORKFLOW_ISSUE_TEMPLATE.md]: > Should we assign the issue instead/as well as mention?

We can't assign to teams, only to individuals... and doing so would mean a lot more code to extract individuals and do some sort of round-robin assignment, etc. Unless you have a concrete idea on how to decide this, I think we can punt for now and decide later if it's worth the effort?

File: .github/WORKFLOW_ISSUE_TEMPLATE.md (hunk 2 above)
Comment (danielrozenberg) [reply .github/WORKFLOW_ISSUE_TEMPLATE.md:1]: `.yml` templates are available for users to create issues with, which is not what we want here. The documentation for https://github.com/JasonEtco/create-an-issue recommends using a .md file in this case. I can try changing this to a .yml file and see if I can exclude it from the create-an-issue page if you feel strongly about this!

**Comment by danielrozenberg** [comment]
> Can you create an issue for the cherry-pick workflow too?

Done 6d80075

> edit: that was surprisingly less code than I expected! nice :)

ikr??? Thanks @JasonEtco 🥳

File: .github/WORKFLOW_ISSUE_TEMPLATE.md (hunk 2 above)
Comment (estherkim) [reply .github/WORKFLOW_ISSUE_TEMPLATE.md:1]: I just feel strongly about not confusing this with a GitHub issue template. You prob don't need to capitalize the whole name. Suggestion - rename to `.github/create_issue_on_error.md`

**Review by estherkim — APPROVED** [review]

File: .github/WORKFLOW_ISSUE_TEMPLATE.md (hunk 2 above)
Comment (danielrozenberg) [reply .github/WORKFLOW_ISSUE_TEMPLATE.md]: Done!

**Pull request merged**